python/ray/ml/predictors/integrations/huggingface/huggingface_predictor.py

```diff
@@ -0,0 +1,159 @@
+import os
+from typing import Optional, Type, Union, List
+
+import numpy as np
+import pandas as pd
+import torch
+from transformers.modeling_utils import PreTrainedModel
+from transformers.trainer import WEIGHTS_NAME
+
+import ray.cloudpickle as cpickle
+from ray.ml.predictor import Predictor, DataBatchType
+from ray.ml.preprocessor import Preprocessor
+from ray.ml.checkpoint import Checkpoint
+from ray.ml.utils.torch_utils import load_torch_model, convert_pandas_to_torch_tensor
+from ray.ml.constants import PREPROCESSOR_KEY
+
+
+class HuggingFacePredictor(Predictor):
```

Comment:
Would it be possible to subclass TorchPredictor to avoid some of the duplicate code?

Reply:
All the methods will have to be overridden regardless.

Reply:
Sure, but we can use super calls to avoid some of the duplicate code.